lib/plugins/aws/provider/awsProvider.js

```diff
@@ -132,6 +132,13 @@ class AwsProvider {
     this.serverless.setProvider(constants.providerName, this);
     if (this.serverless.service.provider.name === 'aws') {
       this.serverless.service.provider.region = this.getRegion();
+      serverless.configSchemaHandler.schema.properties.provider.additionalProperties = true;
+
+      serverless.configSchemaHandler.schema.properties.functions.patternProperties[
+        '^[a-zA-Z0-9-_]+$'
+      ].additionalProperties = true;
+
+      serverless.configSchemaHandler.defineProvider('aws');
```

Comment:
This makes sense.  Provider and function properties should be included in `defineProvider` 
method because that's what this method does by its name.

I'll adjust the `ConfigSchemaHandler` class to make this work